packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -324,25 +324,25 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
   };
 
   generateResolvers = (context: TransformerContextProvider): void => {
-    const dataSourceMapping = {};
+    const dataSourceMapping: Record<string, string> = {};
     this.resourceGeneratorMap.forEach((generator) => {
       generator.generateResources(context);
-      Object.assign(
-        dataSourceMapping,
-        Object.fromEntries(
-          Object.entries(generator.getDatasourceMap())
-            .filter(([, datasource]) => datasource.ds.type === 'AMAZON_DYNAMODB')
-            .map(([modelName, datasource]) => {
-              if (datasource.ds.dynamoDbConfig && !cdk.isResolvableObject(datasource.ds.dynamoDbConfig)) {
-                return [modelName, datasource.ds.dynamoDbConfig.tableName];
-              }
-              throw new Error(`Unable to construct data source mapping. Could not resolve table name for ${modelName}`);
-            }),
-        ),
+      const ddbDatasources = Object.entries(generator.getDatasourceMap()).filter(
+        ([, datasource]) => datasource.ds.type === 'AMAZON_DYNAMODB',
       );
+      ddbDatasources.forEach(([modelName, datasource]) => {
+        if (datasource.ds.dynamoDbConfig && !cdk.isResolvableObject(datasource.ds.dynamoDbConfig)) {
+          dataSourceMapping[modelName] = datasource.ds.dynamoDbConfig.tableName;
+        }
+        // TODO: probably need a link to docs for this
+        console.warn(
+          `Could not resolve table name for ${modelName}. DataSourceMappingOutput is incomplete. Please manually add ${modelName} to the mapping for your migration.`,
+        );
+      });
     });
     if (context.transformParameters.enableGen2Migration && context.transformParameters.enableTransformerCfnOutputs) {
       const { scope } = context.stackManager;
+      // TODO: decide final naming before merge to main
```

Comment:
Note: it's actually the `exportName` that we care about, since that is how customers grab it in their **backend.ts** files.